daemon/cmd/kube_proxy_replacement.go

```diff
@@ -215,7 +210,77 @@ func initKubeProxyReplacementOptions() (bool, error) {
+		if option.Config.TunnelingEnabled() &&
+			option.Config.NodePortMode != option.NodePortModeSNAT {
+
+			log.Warnf("Disabling NodePort's %q mode feature due to tunneling mode being enabled",
+				option.Config.NodePortMode)
+			option.Config.NodePortMode = option.NodePortModeSNAT
+		}
+
+		if option.Config.NodePortAcceleration != option.NodePortAccelerationDisabled &&
+			option.Config.TunnelingEnabled() {
+
+			return false, fmt.Errorf("Cannot use NodePort acceleration with tunneling. Either run cilium-agent with --%s=%s or --%s=%s",
+				option.NodePortAcceleration, option.NodePortAccelerationDisabled, option.TunnelName, option.TunnelDisabled)
+		}
+
+		if option.Config.NodePortMode == option.NodePortModeDSR &&
+			option.Config.LoadBalancerDSRDispatch == option.DSRDispatchIPIP {
+			if option.Config.DatapathMode != datapathOption.DatapathModeLBOnly {
+				return false, fmt.Errorf("DSR dispatch mode %s only supported for --%s=%s", option.Config.LoadBalancerDSRDispatch, option.DatapathMode, datapathOption.DatapathModeLBOnly)
+			}
+			if option.Config.NodePortAcceleration == option.NodePortAccelerationDisabled {
+				return false, fmt.Errorf("DSR dispatch mode %s currently only available under XDP acceleration", option.Config.LoadBalancerDSRDispatch)
+			}
+		}
+
+		option.Config.EnableHealthDatapath =
+			option.Config.DatapathMode == datapathOption.DatapathModeLBOnly &&
+				option.Config.NodePortMode == option.NodePortModeDSR &&
+				option.Config.LoadBalancerDSRDispatch == option.DSRDispatchIPIP
+	}
+
+	if option.Config.InstallNoConntrackIptRules {
+		// InstallNoConntrackIptRules can only be enabled when Cilium is
+		// running in full KPR mode as otherwise conntrack would be
+		// required for NAT operations
+		if !option.Config.KubeProxyReplacementFullyEnabled() {
+			return false, fmt.Errorf("%s requires the agent to run with %s=%s.",
+				option.InstallNoConntrackIptRules, option.KubeProxyReplacement, option.KubeProxyReplacementStrict)
+		}
+
+		if option.Config.MasqueradingEnabled() && !option.Config.EnableBPFMasquerade {
+			return false, fmt.Errorf("%s requires the agent to run with %s.",
+				option.InstallNoConntrackIptRules, option.EnableBPFMasquerade)
+		}
+	}
+
+	if option.Config.BPFSocketLBHostnsOnly {
+		if !option.Config.EnableSocketLB {
+			option.Config.BPFSocketLBHostnsOnly = false
+			log.Warnf("%s only takes effect when %s is true", option.BPFSocketLBHostnsOnly, option.EnableSocketLB)
+		} else if probes.HaveProgramHelper(ebpf.CGroupSockAddr, asm.FnGetNetnsCookie) != nil {
+			option.Config.BPFSocketLBHostnsOnly = false
+			log.Warn("Without network namespace cookie lookup functionality, BPF datapath " +
+				"cannot distinguish root and non-root namespace, skipping socket-level " +
+				"loadbalancing will not work. Istio routing chains will be missed. " +
+				"Needs kernel version >= 5.7")
+		}
+	}
+
+	if option.Config.DryMode {
+		return strict, nil
+	} else {
+		return probeKubeProxyReplacementOptions(strict)
+	}
```

Comment:
nit: simplify the nested return
```suggestion
	if option.Config.DryMode {
		return strict, nil
	}
	return probeKubeProxyReplacementOptions(strict)
```